File: aimova/Models/Shot.swift
import Foundation

enum ShotShape: String, Codable, CaseIterable {
    case fade = "FADE"
    case draw = "DRAW"
    case straight = "STRAIGHT"

    var displayName: String {
        switch self {
        case .fade: return "Fade"
        case .draw: return "Draw"
        case .straight: return "Straight"
        }
    }

    var abbreviation: String {
        switch self {
        case .fade: return "F"
        case .draw: return "D"
        case .straight: return "S"
        }
    }
}

struct ShotCounts: Codable {
    var fade: Int
    var draw: Int
    var straight: Int

    static let empty = ShotCounts(fade: 0, draw: 0, straight: 0)

    enum CodingKeys: String, CodingKey {
        case fade = "FADE"
        case draw = "DRAW"
        case straight = "STRAIGHT"
    }

    var total: Int { fade + draw + straight }

    var displayString: String {
        guard total > 0 else { return "" }
        var parts: [String] = []
        if fade > 0 { parts.append("F:\(fade)") }
        if straight > 0 { parts.append("S:\(straight)") }
        if draw > 0 { parts.append("D:\(draw)") }
        return parts.joined(separator: "  ")
    }
}

struct Shot: Codable, Identifiable {
    let id: String
    let clubId: String
    let userId: String
    var shotShape: ShotShape
    var carryYards: Double
    var offlineYards: Double
    var totalYards: Double?
    var notes: String?
    var loggedAt: Date
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case clubId = "club_id"
        case userId = "user_id"
        case shotShape = "shot_shape"
        case carryYards = "carry_yards"
        case offlineYards = "offline_yards"
        case totalYards = "total_yards"
        case notes
        case loggedAt = "logged_at"
        case createdAt = "created_at"
    }
}

File: backend/models/shot.py
import uuid
from datetime import datetime
from sqlalchemy import String, Float, Text, DateTime, Enum, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column
from database import Base
import enum


class ShotShape(str, enum.Enum):
    FADE = "FADE"
    DRAW = "DRAW"
    STRAIGHT = "STRAIGHT"


class Shot(Base):
    __tablename__ = "shots"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    club_id: Mapped[str] = mapped_column(String(36), ForeignKey("clubs.id"), nullable=False)
    user_id: Mapped[str] = mapped_column(String(36), ForeignKey("users.id"), nullable=False)
    shot_shape: Mapped[ShotShape] = mapped_column(Enum(ShotShape), nullable=False)
    carry_yards: Mapped[float] = mapped_column(Float, nullable=False)
    offline_yards: Mapped[float] = mapped_column(Float, nullable=False)
    total_yards: Mapped[float | None] = mapped_column(Float, nullable=True)
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    logged_at: Mapped[datetime] = mapped_column(DateTime, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
